steps/repositories-describe/step.py

```diff
@@ -25,13 +30,28 @@
 except:
   direction = "asc"
 
+try:
+  fields = relay.get(D.fields)
+except:
+  fields = []
+
 gh = Github(token)
 
 try:
-  repos = gh.get_organization(org).get_repos(type=scope, sort=sort, direction=direction)
-  repos = list(map(lambda x: x.raw_data, repos))
+  if name:
+    repo = gh.get_organization(org).get_repo(name)
+    relay.outputs.set("repository", repo.raw_data)
+  else:
+    repos = gh.get_organization(org).get_repos(type=scope, sort=sort, direction=direction)
+    repos = list(map(lambda x: x.raw_data, repos))
+
+    # ensure that the name is part of the fields; doesn't matter if it's there twice
+    fields.insert(0, 'name')
+
+    # now filter to have only the fields requested so that we don't exceed payload size when saving output.
```

Comment:
I think this should probably be an opt-in filter, i.e., if it's not specified you get all fields instead of none. What do you think?